compiler/rustc_middle/src/hir/mod.rs

```diff
@@ -28,6 +28,7 @@ pub struct ModuleItems {
     trait_items: Box<[TraitItemId]>,
     impl_items: Box<[ImplItemId]>,
     foreign_items: Box<[ForeignItemId]>,
+    opaques: Box<[LocalDefId]>,
```

Comment:
Why is it still necessary to "partially outline" opaques?
Is it a temporary measure to make migration easy, or it's actually useful to visit "all opaques in a module" sometimes?
We don't keep a similar list for closures for example.

Reply:
We do have `body_owners`, which contains both fns, closures and coroutines.

Reply:
I finished reading the PR and saw the places where `opaques` are used, they are all reasonable.